src/applications/ivc-champva/10-7959f-1/config/prefillTransformer.js

```diff
@@ -0,0 +1,26 @@
+export const prefillTransformer = (pages, formData, metadata) => {
+  return {
+    pages,
+    formData: {
+      veteran: {
+        dateOfBirth: formData.veteranDOB,
+        fullName: formData.fullName,
+        first: formData.fullName.first,
+        middle: formData.fullName.middle,
+        last: formData.fullName.last,
+      },
+      physicalAddress: {
+        street: formData.physicalAddress.street,
+        street2: formData.physicalAddress.street2,
+        city: formData.physicalAddress.city,
+        state: formData.physicalAddress.state,
+        country: formData.physicalAddress.country,
+        postalCode: formData.physicalAddress.postalCode,
+      },
+      veteranSocialSecurityNumber: formData.ssn,
+      phoneNumber: formData.phoneNumber,
+      emailAddress: formData.emailAddress,
+    },
```

Comment:
I'm not sure, but I get you're right. I'll go ahead and change the object and we can see if that does something funny.

Reply:
I think this changed to:
```
veteranFullName: [identity_information, full_name]
veteranAddress: [contact_information, address]
veteranDateOfBirth: [identity_information, date_of_birth]
veteranSocialSecurityNumber: [identity_information, ssn]
veteranPhoneNumber: [contact_information, us_phone]
veteranEmailAddress: [contact_information, email]
```

Reply:
@balexandr my bad, I updated it!
@michaelclement it refused to accept ...formdata. Throws an error everytime. Ideas?

Reply:
Hmm - are you doing something like this:
```javascript
export function prefillTransformer(pages, formData, metadata) {
  return {
    pages,
    formData: {
      ...formData,
      veteran: {
        dateOfBirth: formData.veteranDOB,
      },
    },
    metadata,
  };
}
```
Should give you the combined formData object, something like:
![js](https://github.com/department-of-veterans-affairs/vets-website/assets/18408628/915273c8-ae26-4641-bd0b-9418e17fb5f4)

Reply:
```javascript
function prefillTransformer(pages, formData, metadata) {
  return {
    pages,
    formData: {
      // Preserves all original `formData` info, which would include SIP stuff
      ...formData,
      veteran: {
        // This would be the selective reformatting of pre-fill info from the profile
        dateOfBirth: formData.veteranDOB,
      },
    },
    metadata,
  };
}
```

So should give something like this:

![js](https://github.com/department-of-veterans-affairs/vets-website/assets/18408628/2578d919-c84c-4ff4-b4f9-c4e2438eced2)